src/basic/time-util.c

```diff
@@ -1604,15 +1605,73 @@ bool clock_supported(clockid_t clock) {
         }
 }
 
+int find_localtime_last_symlink(char **ret) {
+        _cleanup_free_ char *path = NULL, *last_symlink = NULL;
+        int r;
+        unsigned i;
+
+        assert(ret);
+
+        path = strdup("/etc/localtime");
+        if (!path)
+                return -ENOMEM;
+
+        last_symlink = strdup(path);
+        if (!last_symlink)
+                return -ENOMEM;
+
+        for (i = 0; i < CHASE_MAX; ++i) {
+                _cleanup_free_ char *next = NULL;
+                struct stat st;
+
+                r = lstat(path, &st);
+                if (r == 0 && S_ISLNK(st.st_mode)) {
+                        char *copy = strdup(path);
+                        if (!copy)
+                                return -ENOMEM;
+                        free_and_replace(last_symlink, copy);
+                }
+
+                r = chase(path, /* root= */ NULL, CHASE_STEP|CHASE_NONEXISTENT, &next, /* ret_fd= */ NULL);
+                if (r < 0)
+                        return r;
+
+                if (r == 0) {
+                        char *e = PATH_STARTSWITH_SET(next, "/usr/share/zoneinfo/");
+                        if (e) {
+                                break ;
+                        }
+                } else {
+                        /* r > 0
```

Comment:
copypasta?

Reply:
No, it was not obvious this was always true in my opinion. So I kept it in the comment. But I can remove it.

Reply:
ah ok, sure I don't mind